pallets/inactivity-tracking/src/lib.rs

```diff
@@ -304,6 +319,60 @@ pub mod pallet {
             }
         }
 
+        /// Internal function to populate the current session active collator records with collators
+        /// part of inactive chains.
+        pub fn process_inactive_chains_for_session() {
+            match <CurrentActivityTrackingStatus<T>>::get() {
+                ActivityTrackingStatus::Disabled { .. } => return,
+                ActivityTrackingStatus::Enabled { start, end: _ } => {
+                    if start > T::CurrentSessionIndex::session_index() {
+                        return;
+                    }
+                }
+            }
+            let mut active_chains = <ActiveContainerChainsForCurrentSession<T>>::get().into_inner();
+            // Fetching the parathreads for the current session and remove them
+            // from the active chains array. In this way we handle all parathreads
+            // as inactive chains. This solution would only work if a collator either:
+            // - is assigned to one chain only
+            // - is assigned to multiple chains but all of them are parathreads
+            let parathreads_for_session = T::ParaFilter::get_parathreads_for_session();
+            active_chains.retain(|x| !parathreads_for_session.contains(x));
+
+            let _ = <ActiveCollatorsForCurrentSession<T>>::try_mutate(
+                |active_collators| -> DispatchResult {
+                    let container_chains_with_collators =
+                        T::CurrentCollatorsFetcher::container_chains_with_collators(
+                            ForSession::Current,
+                        );
+
+                    for (para_id, collator_ids) in container_chains_with_collators.iter() {
+                        if !active_chains.contains(para_id) {
+                            // Collators assigned to inactive chain are added
+                            // to the current active collators storage
+                            for collator_id in collator_ids {
+                                if !active_collators.contains(&collator_id) {
```

Comment:
Do we really need this `contains` here?
`active_collators.try_insert` already should raise a error in case of either the limit is reached or collator_id in the set.

Reply:
True, don't think we need it. Removed it in https://github.com/moondance-labs/tanssi/pull/899/commits/f36a655b77148031f869899078a1c4c1b4ed91f3